src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs

```diff
@@ -23,15 +23,19 @@ public static T GetEditorConfigOptionValue<T>(this AnalyzerConfigOptions analyze
 
         public static bool TryGetEditorConfigOption<T>(this AnalyzerConfigOptions analyzerConfigOptions, IOption2 option, out T value)
         {
+            Contract.ThrowIfFalse(option.OptionDefinition.IsEditorConfigOption);
+
             if (typeof(T) == typeof(NamingStylePreferences) && StructuredAnalyzerConfigOptions.TryGetStructuredOptions(analyzerConfigOptions, out var structuredOptions))
             {
                 var preferences = structuredOptions.GetNamingStylePreferences();
                 value = (T)(object)preferences;
                 return !preferences.IsEmpty;
             }
 
+            Contract.ThrowIfNull(option.StorageLocation);
+
             if (analyzerConfigOptions.TryGetValue(option.OptionDefinition.ConfigName, out var stringValue) &&
-               ((EditorConfigStorageLocation<T>)option.StorageLocations.Single()).TryParseValue(stringValue, out value!))
+               ((EditorConfigStorageLocation<T>)option.StorageLocation).TryParseValue(stringValue, out value!))
```

Comment:
do you still need the cast?

Reply:
Changed in subsequent commit: we still cast as an optimization to avoid boxing.